scripts/pysa_fuzzer/fuzzer2.py

```diff
@@ -0,0 +1,317 @@
+from typing import List
+import random
+import textwrap
+import itertools
+import string
+
+class CodeGenerator:
+    def __init__(self) -> None:
+        self.variables = self.generate_variable_names()
+        self.current_var = 0
+    
+    def generate_variable_names(self) -> List[str]:
+        single_letter_names = list(string.ascii_lowercase)
+        two_letter_names = [''.join(pair) for pair in itertools.product(string.ascii_lowercase, repeat=2)]
+        names = single_letter_names + two_letter_names
+        # Remove reserved keywords
+        reserved_keywords = {'as', 'in', 'if', 'is', 'or'}
+        names = [name for name in names if name not in reserved_keywords]
+        return names
+
+    def generate_new_variable(self) -> str:
+        index = self.current_var
+        self.current_var += 1
+        return self.variables[index]
+
+    def get_last_variable(self) -> str:
+        return self.variables[self.current_var - 1]
+
+    def reset(self) -> None:
+        self.current_var = 0
+
+    def generate_import_statements(self) -> str:
+        return "import random\nimport math\n"
+    
+    def generate_source(self) -> str:
+        import_statements = self.generate_import_statements()

```

Comment:
I would call this directly from `generate_statements`. There is no strong reason to have it here.